monai/transforms/utility/dictionary.py

```diff
@@ -1740,6 +1742,68 @@ def __call__(self, data: Mapping[Hashable, NdarrayOrTensor]) -> dict[Hashable, N
         return d
 
 
+class ApplyTransformToPointsd(MapTransform, InvertibleTransform):
+    """
+    Dictionary-based wrapper of :py:class:`monai.transforms.ApplyTransformToPoints`.
+
+    Args:
+        keys: keys of the corresponding items to be transformed.
+            See also: monai.transforms.MapTransform
+        refer_key: the key of the reference item used for transformation.
+        dtype: The desired data type for the output.
+        affine: A 3x3 or 4x4 affine transformation matrix applied to points. Typically, this matrix originates from the image.
+        invert_affine: Whether to invert the affine transformation matrix applied to the points. Defaults to ``True``.
+            Typically, the affine matrix is derived from the image, while the points are in world coordinates.
+            If you want to align the points with the image, set this to ``True``. Otherwise, set it to ``False``.
+        affine_lps_to_ras: Defaults to ``False``. Set this to ``True`` if all of the following are true:
+            1) The image is read by `ITKReader`,
+            2) The `ITKReader` has `affine_lps_to_ras=True`,
+            3) The data is in world coordinates.
+            This ensures the correct application of the affine transformation between LPS (left-posterior-superior)
+            and RAS (right-anterior-superior) coordinate systems. This argument ensures the points and the affine
+            matrix are in the same coordinate system.
+        allow_missing_keys: Don't raise exception if key is missing.
+    """
+
+    def __init__(
+        self,
+        keys: KeysCollection,
+        refer_key: str | None = None,
+        dtype: DtypeLike | torch.dtype = torch.float64,
+        affine: torch.Tensor | None = None,
+        invert_affine: bool = True,
+        affine_lps_to_ras: bool = False,
+        allow_missing_keys: bool = False,
+    ):
+        MapTransform.__init__(self, keys, allow_missing_keys)
+        self.refer_key = refer_key
+        self.affine = affine
+        if self.refer_key is None and self.affine is None:
+            warnings.warn("No reference data or affine matrix is provided, will use the affine derived from the data.")
+        self.converter = ApplyTransformToPoints(
+            dtype=dtype, invert_affine=invert_affine, affine_lps_to_ras=affine_lps_to_ras
+        )
+
+    def __call__(self, data: Mapping[Hashable, torch.Tensor]):
+        d = dict(data)
+        refer_data = d[self.refer_key] if self.refer_key is not None else None
+        if isinstance(refer_data, MetaTensor):
+            affine = refer_data.affine
+        else:
+            warnings.warn("No reference affine find in the refer key, will use the affine derived from the data.")
+            affine = self.affine
```

Comment:
Hi @ericspod, good point, I updated it to be `affine = getattr(refer_data, "affine", refer_data)`, and also update the docstring for the `refer_key` to be:
```
        refer_key: The key of the reference item used for transformation.
            It can directly refer to an affine or an image from which the affine can be derived.
```
Do you think this good now?

Reply:
I think this is good, thanks!